docs/new_flux_ideas_and_guide.md

```diff
@@ -0,0 +1,985 @@
+
+The following functions can be used in the **Selector Predicate Block**: `first`, `last`, `min`, `max`, `top`, `bottom`
+
+#### Calling Aggregates without the Aggregate function
+The `aggregate` function provides a shorthand for computing multiple aggregates at a time. However, the 
+functions passed in can also be called on tables. For example:
+
+```
+// get the most recently written record
+@my_data.h2o_temperature |> last()
+
+// get the most recently written record for each location
+@my_data.h2o_temperature |> last(by: ["location"])
+
+// get the average for each hour for each location
+@my_data.h2o_temperature{time > -3h} |> mean(by: ["location"], window: 1h)
+``` 
+
+Selectors can only be called with the `select` function.
+
+#### Count, Distinct, and CountDistinct
+The `count` function is an aggregate while the `distinct` function is a selector. The `countDistinct` function is 
+a helper function that will combine operations for you.
+ 
+```
+@my_data.h2o_temperature{state, time > -3h}
+    |> select(fn: distinct(state), window: 1h, windowColumn: "time")
+```
+
+Would return the following result
+
+| state | time
+| --- | --- |
+| CA | 2020-02-22T16:00
+| CA | 2020-02-22T17:00
+| CA | 2020-02-22T18:00
+| WA | 2020-02-22T18:00
+
+We can see that CA reported in all three hours while both CA and WA reported in only the last hour. Also see that 
+distinct discards columns that aren't in the `by` argument or the `distinct` column itself. So we use the `time` 
+name for where we put the window markers for each distinct state. You could then count the number of states reporting 
+per hour:
+
+```
+@my_data.h2o_temperature{state, time > -3h}
+    |> select(fn: distinct(state), window: 1h)
+    |> timeShift(-1m)
+    |> count("state", window: 1h)
+```
+
+Would produce:
+
+| state | time
+| --- | --- |
+| 1 | 2020-02-22T16:00
+| 1 | 2020-02-22T17:00
+| 2 | 2020-02-22T18:00
+
+See that we had to use the `timeShift` function to move the timestamps of the windows back so that the resulting 
+times would be accurate for `aggregate`.
+
+All of this is encapsulated by the `countDistinct` function. For example, if we wanted to count the number of 
+distinct locations that reported in each sate for each hour in the last three hours.
+
+```
+@my_data.h2o_temperature{location, state, bottom_degrees, time > -3h}
+    |> countDistinct(location, by: ["state"], window: 1h)
+```
+
+### Math, basic transformations, and using previous results in new queries
+Flux makes it easy to do basic math and transformations across tables of data. First, let's look at 
+renaming columns. All of the different predicate blocks recognize the `as` keyword to rename. For 
+example:
+
+```
+@my_data.h2o_temperature{location, state, bottom_degrees as degrees, top_degrees, time}
+
+// or renaming columns in an aggregate:
+@my_data.h2o_temperature{bottom_degrees, time > -1h}
+    |> aggregate({min(bottom_degrees) as min, max(bottom_degrees) as max})
```

Comment:
I really dislike that without a pipe forward I get an implicit last. But with a pipe I get all.

Reply:
No, that's not the case at all. Without the `time > -1h` part in the filter predicate you get `last`. Otherwise you get values in that range.

Reply:
Are there a lot of use cases where the user would want `last` by default? I'm thinking maybe in a task, but its hard for me to think of other times where this would be the preferred default behavior.

Reply:
Curious about this too, what is the rationale behind using last? Is it simply something that isn't *everything* when they neglect to give a time constraint. In other words, a *sane* default?